kornia/geometry/line.py

```diff
@@ -0,0 +1,173 @@
+            p0: tensor with first point :math:`(B, D)`.
+            p1: tensor with second point :math:`(B, D)`.
+
+        Example:
+            >>> p0 = torch.tensor([0.0, 0.0])
+            >>> p1 = torch.tensor([1.0, 1.0])
+            >>> l = ParametrizedLine.through(p0, p1)
+        """
+        return ParametrizedLine(p0, normalize((p1 - p0), p=2, dim=-1))
+
+    @classmethod
+    def from_hyperplane(cls, plane: Hyperplane) -> "ParametrizedLine":
+        raise NotImplementedError(f"Plane not implemented yet {plane}.")
+
+    def point_at(self, t: Union[float, Tensor]) -> Tensor:
+        """The point at :math:`t` along this line.
+
+        Args:
+            t: step along the line.
+
+        Return:
+            tensor with the point.
+
+        Example:
+            >>> p0 = torch.tensor([0.0, 0.0])
+            >>> p1 = torch.tensor([1.0, 1.0])
+            >>> l = ParametrizedLine.through(p0, p1)
+            >>> p2 = l.point_at(0.1)
+        """
+        return self.origin + self.direction * t
+
+    def projection(self, point: Tensor) -> Tensor:
+        """Return the projection of a point onto the line.
+
+        Args:
+            point: the point to be projected.
+        """
+        return self.origin + (self.direction @ (point - self.origin)) * self.direction
+
+    def squared_distance(self, point: Tensor) -> Tensor:
+        """Return the squared distance of a point to its projection onte the line.
+
+        Args:
+            point: the point to calculate the distance onto the line.
+        """
+        diff: Tensor = point - self.origin
+        return squared_norm(diff - (self.direction @ diff) * self.direction)
+
+    def distance(self, point: Tensor) -> Tensor:
+        """Return the distance of a point to its projections onto the line.
+
+        Args:
+            point: the point to calculate the distance into the line.
+        """
+        return self.squared_distance(point).sqrt()
+
+    # TODO(edgar) implement the following:
+    # - intersection
+    # - intersection_parameter
+    # - intersection_point
+
+
+def fit_line(points: Tensor, weights: Optional[Tensor] = None) -> Tensor:
+    """Fit a line from a set of points.
+
+    Args:
+        points: tensor containing a batch of sets of n-dimensional points. The  expected
+            shape of the tensor is :math:`(B,N,D)`.
+        weights: weights to use to solve the equations system. The  expected
+            shape of the tensor is :math:`(B,N)`.
+
+    Return:
+        A tensor containing the direction of the fited line of shape :math:`(B,D)`.
+
+    Example:
+        >>> points = torch.rand(2, 10, 3)
+        >>> weights = torch.ones(2, 10)
+        >>> direction = fit_line(points, weights)
+        >>> direction.shape
+        torch.Size([2, 3])
+    """
+    KORNIA_CHECK_IS_TENSOR(points, "points must be a tensor")
+    KORNIA_CHECK_SHAPE(points, ["B", "N", "D"])
+
+    points_mean = points.mean(-2, True)
+    A = points - points_mean
```

Comment:
I do not remember if this substaraction is mandatory but as far as I know you are changing teh line equation so some op should be done at the end. If you only return the "direction" then you are not fitting a line.

Reply:
probably we should return the mean too

Reply:
actually lets just return a parametrized line